### PR TITLE
feat(otel): add otel to ingest API

### PR DESCRIPTION
**Pull request by ividito**
### Issue

#289 

### What?

- Adds basic monitoring functionality to ingest API

### Testing?

- Deployed to temporary `otel` stack and viewed generated metrics and logs using POC Grafana instance

### CODE REVIEW

File: ingest_api/runtime/src/main.py [hunk 1]
```diff
@@ -222,7 +227,42 @@ def who_am_i(claims=Depends(auth.decode_token)):
...
 async def validation_exception_handler(request, exc):
     return JSONResponse(str(exc), status_code=422)
+
+
+@app.exception_handler(Exception)
+async def general_exception_handler(request, err):
+    """Handle exceptions that aren't caught elsewhere"""
+    metrics.add_metric(name="UnhandledExceptions", unit=MetricUnit.Count, value=1)
+    logger.exception("Unhandled exception")
+    return JSONResponse(status_code=500, content={"detail": "Internal Server Error"})
```
Comment (anayeaye) [inline ingest_api/runtime/src/main.py:268]: Can we log the err here? `f"Unhandled exception {err=}"`?

**Review by anayeaye — APPROVED** [review]
Approve but I am curious about the unhandled/unlogged exception err.

**Comment by anayeaye** [comment]
@smohiudd @ividito I wonder if we should push this through before the auth PR #356 (we also may have to update the auth branch to resolve any diffs).

File: ingest_api/runtime/src/main.py (hunk 1 above)
Comment (smohiudd) [reply ingest_api/runtime/src/main.py:268]: I think it makes sense to log the error

**Pull request merged**